src/SonarScanner.MSBuild.Shim/PathHelper.cs

```diff
@@ -78,5 +87,28 @@ public static string[] GetParts(DirectoryInfo directory)
             parts.Add(directory.Name);
             return parts.AsEnumerable().Reverse().ToArray();
         }
+
+        private static string BestRoot(string[][] pathParts)
```

Comment:
this is more like "BestDrive", right? 

Because "root" can be a folder, too, but this starts with `pathParts.Select(x => x[0])`

Reply:
Thinking of unix file system, it's not a drive but a root of the path.
```
/home/me/project/src
/home/me/project/test
/var/something
/tmp/
```

Reply:
we are using `root` with two different meanings in the file:
- "common root" which can be `/home/me/project/test`
- "BestRoot" which will only be the first element of the path array, from what I understand in the code (either `C:` or `home`) 

Can we find a way to differentiate between these two meanings?

Reply:
I like `BestRoot`, because it's really a root. Let's rename `BestCommonRoot` and `ComputeRootProjectBaseDir`.

`BestCommonPrefix` and `ComputeProjectBaseDir` (as this one doesn't have to come from common prefix but other sources and parameters)?

Reply:
I agree, your solution removes the ambiguity on what "root" means.